ublox_derive/src/output.rs

```diff
@@ -635,30 +716,69 @@ pub fn generate_code_for_parse(recv_packs: &RecvPackets) -> TokenStream {
         #[doc = "All possible packets enum"]
         #[derive(Debug)]
-        pub enum #union_enum_name<'a> {
-            #(#pack_enum_variants),*,
-            Unknown(#unknown_var<'a>)
+        pub enum #union_enum_name_ref<'a> {
+            #(#pack_enum_variants_ref),*,
+            Unknown(#unknown_var_ref<'a>)
+        }
+        #[doc = "All possible packets enum, owning the underlying data"]
+        #[derive(Debug)]
+        pub enum #union_enum_name_owned {
+            #(#pack_enum_variants_owned),*,
+            Unknown(#unknown_var_owned)
         }
 
-        impl<'a> #union_enum_name<'a> {
+        impl<'a> #union_enum_name_ref<'a> {
             pub fn class_and_msg_id(&self) -> (u8, u8) {
                 match *self {
-                    #(#class_id_matches),*,
-                    #union_enum_name::Unknown(ref pack) => (pack.class, pack.msg_id),
+                    #(#class_id_matches_ref),*,
+                    #union_enum_name_ref::Unknown(ref pack) => (pack.class, pack.msg_id),
+                }
+            }
+
+            pub fn to_owned(&self) -> #union_enum_name_owned {
+                self.into()
+            }
+        }
+        impl #union_enum_name_owned {
+            pub fn class_and_msg_id(&self) -> (u8, u8) {
+                match *self {
+                    #(#class_id_matches_owned),*,
+                    #union_enum_name_owned::Unknown(ref pack) => (pack.class, pack.msg_id),
                 }
             }
         }
 
-        pub(crate) fn match_packet(class: u8, msg_id: u8, payload: &[u8]) -> Result<#union_enum_name, ParserError> {
+        pub(crate) fn match_packet(class: u8, msg_id: u8, payload: &[u8]) -> Result<#union_enum_name_ref, ParserError> {
             match (class, msg_id) {
-                #(#matches)*
-                _ => Ok(#union_enum_name::Unknown(#unknown_var {
+                #(#matches_ref)*
+                _ => Ok(#union_enum_name_ref::Unknown(#unknown_var_ref {
                     payload,
                     class,
                     msg_id
                 })),
             }
         }
+        pub(crate) fn match_packet_owned(class: u8, msg_id: u8, payload: &[u8]) -> Result<#union_enum_name_owned, ParserError> {
+            match (class, msg_id) {
+                #(#matches_owned)*
+                _ => Ok(#union_enum_name_owned::Unknown(#unknown_var_owned {
+                    payload: payload.into(),
+                    class,
+                    msg_id
+                })),
+            }
+        }
+
+        impl<'a> From<&#union_enum_name_ref<'a>> for #union_enum_name_owned {
+            fn from(packet: &#union_enum_name_ref<'a>) -> Self {
+                match packet {
+                    #(#matches_ref_to_owned)*
+                    #union_enum_name_ref::Unknown(#unknown_var_ref {payload, class, msg_id}) => {
+                        PacketOwned::Unknown(#unknown_var_owned { payload: payload.to_vec(), class: *class, msg_id: *msg_id })
```

Comment:
I _think_ this is the only place affected by my `alloc` comment above, correct me if I'm wrong.